src/puppetlabs/trapperkeeper/authorization/rules.clj

```diff
@@ -15,6 +15,7 @@
    :type (schema/enum :string :regex)
    :path Pattern
    :method (schema/enum :get :post :put :delete :head :any)
+   (schema/optional-key :allow_unauthenticated) schema/Bool
```

Comment:
For all of the settings that we've been adding to our Clojure projects (Trapperkeeper / Puppet Server), I believe we've been pretty consistent about using a hyphen instead of an underscore in the name.  Would you be okay with changing the references here from `allow_unauthenticated` to `allow-unauthenticated` - both where used as a keyword in the code and as the name of the key represented in the HOCON file loaded from disk?